### PR TITLE
chore(): forbidden identifiers check for relative scope imports.

### PR DESCRIPTION
**Pull request by DevVersion**
* Checks the added imports for any relative scope imports and automatically suggests the correct Scope Import statement.

Once a forbidden import will be found, we will output a **super** helpful error message (including suggestion / correction)

image

cc. @jelbourn

### CODE REVIEW

File: scripts/ci/forbidden-identifiers.js [hunk 1]
```diff
@@ -33,7 +34,16 @@ const blocked_statements = [
   'from \\\'rxjs/Rx\\\'',
 ];
 
+// Retrieves all scope packages dynamically from the source.
+// Developers shouldn't be able to import from other scope packages by using relative paths.
+const conpmponentFolders = fs
```
Comment (Dona278) [inline scripts/ci/forbidden-identifiers.js]: Should be "componentFolders"?

Comment (DevVersion) [inline scripts/ci/forbidden-identifiers.js]: Yeah, good catch - definitely a cool naming convention :)

**Comment by jelbourn** [comment]
@hansl can you review this one?

File: scripts/ci/forbidden-identifiers.js [hunk 2]
```diff
@@ -33,7 +34,16 @@ const blocked_statements = [
...
+// Retrieves all scope packages dynamically from the source.
+// Developers shouldn't be able to import from other scope packages by using relative paths.
+const componentFolders = fs
+  .readdirSync(`${__dirname}/../../src/components`)
+  .map(componentName => `src/components/${componentName}`);
+
+const scopePackages = ['src/core'].concat(componentFolders);
+
 const blockedRegex = new RegExp(blocked_statements.join('|'), 'g');
+const importRegex = /from '(.*)'/g;
```
Comment (hansl) [inline scripts/ci/forbidden-identifiers.js]: This doesn't check if the line is an `import`. What if `from '...'` is put inside a comment?

File: scripts/ci/forbidden-identifiers.js [hunk 3]
```diff
@@ -116,23 +183,35 @@ findChangedFiles()
...
             lineNumber: ln,
-            statement: m[0]
-          });
+            statement: matches && matches[0] || isScopeImport.invalidStatement
+          };
+
+          if (isScopeImport) {
+            error.message =
+              '   You are using an import statement, which imports a file from another scope package.\n' +
+              `   Please import the file by using the following path: ${isScopeImport.scopeImportPath}`;
```
Comment (hansl) [inline scripts/ci/forbidden-identifiers.js]: Please add a `\n` at the end of the message.

**Comment by hansl** [comment]
LGTM with this note. Please fix and I'll merge. Cheers!

File: src/components/checkbox/checkbox.ts [hunk 4]
```diff
@@ -8,10 +8,9 @@ import {
     Provider,
     Renderer,
     ViewEncapsulation,
-    forwardRef,
-    AfterContentInit
-} from '@angular/core';
-import {NG_VALUE_ACCESSOR, ControlValueAccessor} from '@angular/common';
+    forwardRef
+} from '../../core/core';
```
Comment (hansl) [inline src/components/checkbox/checkbox.ts]: ?

Comment (DevVersion) [inline src/components/checkbox/checkbox.ts]: This was just commited for testing purposes and I already force-pushed to remove it :)

**Comment by hansl** [comment]
LGTM!  Let's wait for the tests to pass and I'll merge. Thanks a lot!

**Pull request merged**